specification/programmableconnectivity/resource-manager/Microsoft.ProgrammableConnectivity/preview/2023-11-01-preview/OpenApiGateways.json

```diff
@@ -0,0 +1,745 @@
+      "post": {
+        "operationId": "OpenApiGateways_ResetConnection",
+        "tags": [
+          "OpenApiGateways"
+        ],
+        "description": "A long-running resource action that returns no content. DEPRECATED: Use 'ArmResourceActionNoResponseContentAsync' instead",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v5/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "name": "name",
+            "in": "path",
+            "description": "APC OpenAPI Gateway Name",
+            "required": true,
+            "type": "string",
+            "pattern": "^[a-zA-Z0-9-]{3,24}$"
+          },
+          {
+            "name": "body",
+            "in": "body",
+            "description": "The content of the action request",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/OpenApiGateway"
+            }
+          }
+        ],
+        "responses": {
+          "202": {
+            "description": "Resource operation accepted.",
+            "headers": {
+              "Retry-After": {
+                "type": "integer",
+                "format": "int32",
+                "description": "The Retry-After header can indicate how long the client should wait before polling the operation status."
+              },
+              "Location": {
+                "type": "string",
+                "description": "The Location header contains the URL where the status of the long running operation can be checked."
+              }
+            }
+          },
+          "default": {
+            "description": "An unexpected error response.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v5/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "OpenApiGateways_ResetConnection_MaximumSet_Gen": {
+            "$ref": "./examples/OpenApiGateways_ResetConnection_MaximumSet_Gen.json"
+          },
+          "OpenApiGateways_ResetConnection_MinimumSet_Gen": {
+            "$ref": "./examples/OpenApiGateways_ResetConnection_MinimumSet_Gen.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ProgrammableConnectivity/openApiGateways/{name}/updateCredentials": {
```

Comment:
Is it possible to model the credentials as part of the openApiGateway resource? So it could use the PUT/PATCH openAPiGateway methods to update the credentials?